python/eos/tasks.py

```diff
@@ -79,6 +81,97 @@ def task_wrapper(*args, **kwargs):
+    :type start_point: `numpy.ndarray`, optional
+    :param from_mcmc: If provided, the optimization will start at the point with the largest posterior of previously computed MCMC samples.
+        The samples are expected to be stored in the `mcmc-XXXX` subdirectories of the `base_directory`, where 'XXXX' is the provided int.
+    :type from_mcmc: int, optional
+    :param seed: If provided, the optimization will start from a random point and the random number generator will be seeded with this value.
+    :type seed: int, optional
+    '''
+    if optimizations < 1:
+        raise ValueError('The number of optimizations should be larger than zero.')
+    if seed and not start_point is None:
+        raise ValueError('The arguments seed and start_point are mutually exclusive')
+    if seed and not from_mcmc is None:
+        raise ValueError('The arguments seed and fron_mcmc are mutually exclusive')
+    if not from_mcmc is None and not start_point is None:
+        raise ValueError('The arguments from_mcmc and start_point are mutually exclusive')
+
+    if type(analysis_file) is not eos.AnalysisFile:
+        _analysis_file = eos.AnalysisFile(analysis_file)
+    else:
+        _analysis_file = analysis_file
+    analysis = _analysis_file.analysis(posterior)
+    min_chi2 = sys.float_info.max
+    gof = None
+    bfp = None
+
+    eos.info(f'Starting minimization in {optimizations} points')
+
+    if not start_point is None:
+        starting_point = _np.array(start_point)
+        eos.info('Starting optimization from user-provided point')
+
+        _bfp = analysis.optimize(start_point = starting_point)
+        _gof = eos.GoodnessOfFit(analysis._log_posterior)
+        _chi2 = _gof.total_chi_square()
+    elif from_mcmc >= 0:
+        eos.info('Initializing starting point from MCMC data file')
+        chain = eos.data.MarkovChain(os.path.join(base_directory, posterior, f'mcmc-{from_mcmc:04}'))
+        idx_mode = _np.argmax(chain.weights)
+        for p, v in zip(analysis.varied_parameters, chain.samples[idx_mode]):
+            p.set(v)
+
+        _bfp = analysis.optimize()
+        _gof = eos.GoodnessOfFit(analysis._log_posterior)
+        _chi2 = _gof.total_chi_square()
+    else:
+        eos.info('Starting optimization from a random point')
+        if seed is None:
+            seed = 17
+        _bfp = analysis.optimize(start_point='random', rng=_np.random.mtrand.RandomState(seed))
+        _gof = eos.GoodnessOfFit(analysis._log_posterior)
+        _chi2 = _gof.total_chi_square()
+
+    for i in range(optimizations - 1):
+        starting_point = [float(p) for p in analysis.varied_parameters]
+        _bfp = analysis.optimize(start_point = starting_point)
+        _gof = eos.GoodnessOfFit(analysis._log_posterior)
+        _chi2 = _gof.total_chi_square()
+        if _chi2 < min_chi2:
+            gof = _gof
+            bfp = _bfp
+            min_chi2 = _chi2
+
+    print('best-fit point:')
```

Comment:
Ok, this does not work anymore for a task. Can you convert it to ``info`` log messages?

Reply:
It does when I tested it, but in any case we need to write the bfp somewhere. I will use info.